flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -365,6 +368,38 @@ public static TimestampData toTimestampData(double v, int precision) {
         }
     }
 
+    public static TimestampData toTimestampData(int v, int precision) {
+        switch (precision) {
+            case 0:
+                if (MIN_EPOCH_SECONDS <= v && v <= MAX_EPOCH_SECONDS) {
+                    return timestampDataFromEpochMills((v * MILLIS_PER_SECOND));
+                } else {
+                    return null;
+                }
+            case 3:
+                return timestampDataFromEpochMills(v);
+            default:
+                throw new TableException(
+                        "The precision value '"
+                                + precision
+                                + "' for function "
+                                + "TO_TIMESTAMP_LTZ(numeric, precision) is unsupported,"
+                                + " the supported value is '0' for second or '3' for millisecond.");
+        }
+    }
+
+    public static TimestampData toTimestampData(long epoch) {
+        return toTimestampData(epoch, DEFAULT_PRECISION);
+    }
+
+    public static TimestampData toTimestampData(double epoch) {
+        return toTimestampData(epoch, DEFAULT_PRECISION);
+    }
+
+    public static TimestampData toTimestampData(DecimalData epoch) {
+        return toTimestampData(epoch, DEFAULT_PRECISION);
+    }
```

Comment:
Move this to function class

Reply:
Hi @snuyanzin , thanks for the comment! 

I need to register these functions in `BuiltInMethods.scala` so that they can be exposed API calls. If I move these methods to my function class, I will have to  `lookupMethod` my new function class there. Do you think that’s preferred than the current way? My reasoning for the current way is that `lookupMethod` only need to call the common `DataTimeUtils.java` library class.

Reply:
i wonder why do we still have anything in classes under `flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/` for this function?
If we want to move to new stack then there should be nothing.
IIRC there are only functions written in old stack approach

Reply:
Hi @snuyanzin , you asked a great question. I spent quite some time on this, and I think I figured it out. 

In this ticket, we aim to ensure that the existing Scala tests pass to confirm that the function's behavior remains unchanged, therefore we still depend on some scala auto-generated code, which requires modifications to the Scala folder. If I don't modify these folders, my new java tests fail because they can't pick up the new functions.

I attempted to use the new Java tech stack, but that caused the existing Scala tests to fail. 

Once the new functionality is out and stable, we should definitely complete the migration by removing the Scala tests and fully transition to Java, but now I think it's OK to keep it as is so we have the confidence that we don't break existing tests. What do you think?

Reply:
Can you elaborate a bit more here what tests depend on it?

From my point of view during tests we should use either SQL or table api and both are public so ideally tests should not depend on internal implementation like old/new function stack

Reply:
Got rid of the scala tech stack in the new commit. (address 3rd round comment)